### PR TITLE
[1.x] Vite 5

### PR DESCRIPTION
**Pull request by timacdonald**

### CODE REVIEW

File: UPGRADE.md [hunk 1]
```diff
@@ -1,5 +1,36 @@
...
+
+The following command will update `laravel-vite-plugin` and `vite` only:
+
+```
+npm install vite@^5.0.0 laravel-vite-plugin@^1.x
+```
+
+You should also check the upgrading guide and changelogs for any packages you update. The [`vite` migrations guide](https://vitejs.dev/guide/migration.html) is available on their website.
+
+2. Ensure your `package.json` contains `"type": "modules"`.
```
Comment (shoeyn) [inline UPGRADE.md:24]: Should be `"type": "module"` instead of "modules"

Comment (timacdonald) [reply UPGRADE.md]: Thanks!

**Comment by AntonOfTheWoods** [comment]
Any visibility on when the `1.x` branch might be released?

File: src/index.ts [hunk 2]
```diff
@@ -160,7 +160,7 @@ function resolveLaravelPlugin(pluginConfig: Required<PluginConfig>): LaravelPlug
                             ...serverConfig.hmr,
                             ...(userConfig.server?.hmr === true ? {} : userConfig.server?.hmr),
                         },
-                        https: userConfig.server?.https === false ? false : {
+                        https: typeof userConfig.server?.https !== 'undefined' ? userConfig.server.https : {
                             ...serverConfig.https,
                             ...(userConfig.server?.https === true ? {} : userConfig.server?.https),
```
Comment (Jubeki) [inline src/index.ts:165]: @timacdonald I took a look at the PR and found something which doesn't make sense to me without diving much deeper into the source:

The upgrade guide says that boolean support for `https` is dropped, but there is still a ```=== true``` check here. Is this intended? (I only want to make you aware, I don't know if this is needed or not)

Comment (timacdonald) [reply src/index.ts:165]: Thanks! Fixed that one up.

No idea why typescript isn't complaining about that 😵

**Pull request merged**